btcutil/address_test.go

```diff
@@ -879,6 +879,67 @@ func TestAddresses(t *testing.T) {
 			continue
 		}
 
+		// Ensure addresses are only valid for their given network
```

Comment:
IMO rather than stuff this into the existing test (for the _other_ decoding function), we should make an isolated unit test to exercise the new function. Even just reviewing this fragment, it isn't very easy to follow what exactly is being tested here. A new TDD test would make things a lot clearer and more explicit.